repository/repository.go

```diff
@@ -85,3 +86,16 @@ func (r *inMemoryRepository) GetErrors(serviceName string, numberOfErrors int) (
 	metrics.ServiceErrors.WithLabelValues("service_not_found").Inc()
 	return nil, fmt.Errorf("service %s not found", serviceName)
 }
+
+func (r *inMemoryRepository) DeleteError(serviceName string, key string) {
+	if value, ok := r.AggregatedError.Load(serviceName); ok {
+		value, _ := value.([]ErrorAggregate)
+		newValues := []ErrorAggregate{}
+		for _, errorObj := range value {
```

Comment:
is there a way to delete the error from the sync Map without iterating through the entire list?

Reply:
We need to delete the object from the list, so I don't see any other way to do it. We can also change repository representation to be a Map indexing by aggregation key instead of list